modules/disk-uploader/pkg/parse/clioptions_test.go

```diff
@@ -0,0 +1,71 @@
+package parse_test
+
+import (
+	"fmt"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/modules/disk-uploader/pkg/parse"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap/zapcore"
+)
+
+var _ = Describe("CLIOptions", func() {
+	Context("invalid cli options", func() {
+		DescribeTable("Init return correct assertion errors", func(expectedErrMessage string, options *parse.CLIOptions) {
+			err := options.Init()
+			Expect(err).Should(HaveOccurred())
+			fmt.Println(err.Error())
+			Expect(err.Error()).To(ContainSubstring(expectedErrMessage))
+		},
+			Entry("no export-source-kind", "export-source-kind param has to be specified",
+				&parse.CLIOptions{}),
+			Entry("no export-source-name", "export-source-name param has to be specified",
+				&parse.CLIOptions{ExportSourceKind: "vm"}),
+			Entry("no volume-name", "volume-name param has to be specified",
+				&parse.CLIOptions{ExportSourceKind: "vm", ExportSourceName: "test-vmexport"}),
+			Entry("no image-destination", "image-destination param has to be specified",
+				&parse.CLIOptions{ExportSourceKind: "vm", ExportSourceName: "test-vmexport", VolumeName: "test-volume"}),
+		)
+	})
+
+	Context("valid cli options", func() {
+		DescribeTable("Init should succeed", func(options *parse.CLIOptions) {
+			Expect(options.Init()).To(Succeed())
+		},
+			Entry("with yaml output", &parse.CLIOptions{
+				ExportSourceKind:      "vm",
+				ExportSourceNamespace: "test-namespace",
+				ExportSourceName:      "test-vmexport",
+				VolumeName:            "test-volume",
+				ImageDestination:      "quay.io/kubevirt/example",
+				PushTimeout:           60,
+				Debug:                 true,
+			}),
+		)
+
+		It("Init should trim spaces", func() {
+			options := &parse.CLIOptions{
+				ExportSourceKind:      " " + "vm" + " ",
+				ExportSourceNamespace: " " + "test-namespace" + " ",
+				ExportSourceName:      " " + "test-vmexport" + " ",
+				VolumeName:            " " + "test-volume" + " ",
+				ImageDestination:      " " + "quay.io/kubevirt/example" + " ",
```

Comment:
If you use constants for the parameters, we can be sure there is no typo between values here and in the `Expect()` statements below.